### PR TITLE
document bare metal ipi configuration parameters

### PR DESCRIPTION
**Pull request by dhellmann**
Provide a table describing all of the configuration parameters for IPI
deployments on bare metal. Include a more verbose description of the
libvirtURI parameter and the context in which specifying it is
useful. Include examples of BMC address values.

### CODE REVIEW

**Comment by dhellmann** [comment]
The goal here is to document some of the inputs to the installer that are either not documented or are documented in other repositories, so that we have 1 source describing them.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1507/

**Comment by abhinavdahiya** [comment]
how does this documentation for configuration compare to https://github.com/openshift/installer/blob/0c223467d03724769173ac52bf14936c3336b2f1/docs/user/metal/customization_ipi.md

should most of this go into that document?

**Comment by stbenjam** [comment]
Thanks so much for taking the time to do this. We added the customization_ipi.md @abhinavdahiya mentioned above to conform to other platforms' structure. I think the line between what goes there vs here should be optional vs mandatory info.  Host and BMC can probably stay here, as everyone needs to provide that.

**Comment by dhellmann** [comment]
I moved the content about libvirtURI from install_ipi.md to customize_ipi.md.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1510/

File: docs/user/metal/install_ipi.md [hunk 1]
```diff
@@ -177,6 +177,69 @@ pullSecret: ...
 sshKey: ...
 ```
 
+#### Configuration Parameters
```
Comment (stbenjam) [inline docs/user/metal/install_ipi.md]: I like this table but we're now documenting many parametersin two places.  Most of these are covered in customization_ipi.md, could we put them there, and maybe only define the absolute minimum required here?

File: docs/user/metal/install_ipi.md [hunk 2]
```diff
@@ -177,6 +177,69 @@ pullSecret: ...
...
+
+| Parameter | Default | Description |
+| --- | --- | --- |
+`libvirtURI` | `qemu://localhost/system` | The location of the hypervisor for running the bootstrap VM. See [Using a Remote Hypervisor](customization_ipi.md#using-a-remote-hypervisor) for more details. |
+`clusterProvisioningIP` | The third address on the provisioning network. `172.22.0.3` | The IP within the cluster where the provisioning services run. |
+`bootstrapProvisioningIP` | The second address on the provisioning network. `172.22.0.2` | The IP on the bootstrap VM where the provisioning services run while the control plane is being deployed. |
+`externalBridge` | `baremetal` | The name of the bridge of the hypervisor attached to the external network. |
+`provisioningBridge` | `provisioning` | The name of the bridge on the hypervisor attached to the provisioning network. |
+`provisioningNetworkInterface` | | The name of the network interface on control plane nodes connected to the provisioning network. |
```
Comment (stbenjam) [inline docs/user/metal/install_ipi.md]: This one is always required

File: docs/user/metal/install_ipi.md [hunk 3]
```diff
@@ -177,6 +177,69 @@ pullSecret: ...
...
+`provisioningNetworkCIDR` | `172.22.0.0/24` | The CIDR for the network to use for provisioning. |
+`provisioningDHCPExternal` | | Flag indicating that DHCP for the provisioning network is managed outside of the cluster by existing infrastructure services. |
+`provisioningDHCPRange` | The tenth through the hundredth IP on the provisioning network. `172.22.0.10,172.22.0.100` | The IP range to use for hosts on the provisioning network. |
+`hosts` | | Details about bare metal hosts to use to build the cluster. See [below](#host-properties) for more details. |
+`defaultMachinePlatform` | | The default configuration used for machine pools without a platform configuration. |
+`apiVIP` | `api.<clusterdomain>` | The VIP to use for internal API communication. |
+`ingressVIP` | `test.apps.<clusterdomain>` | The VIP to use for ingress traffic. |
+`dnsVIP` | | The VIP to use for internal DNS communication. |
```
Comment (stbenjam) [inline docs/user/metal/install_ipi.md]: These 3 either have to be specified, or available in DNS. I could be wrong, the network folks might know

File: docs/user/metal/install_ipi.md (hunk 1 above)
Comment (dhellmann) [reply docs/user/metal/install_ipi.md]: I've pushed another patch to move some of the settings as you suggest. Let me know what you think.

File: docs/user/metal/install_ipi.md (hunk 3 above)
Comment (dhellmann) [reply docs/user/metal/install_ipi.md]: I moved them back to this file and added a "VIP Settings" section below.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1515/

**Comment by dhellmann** [comment]
I squashed the commits down to 1 change.

File: docs/user/metal/install_ipi.md [hunk 4]
```diff
@@ -177,6 +177,65 @@ pullSecret: ...
...
+`hosts` | | Details about bare metal hosts to use to build the cluster. See below for more details. |
+`defaultMachinePlatform` | | The default configuration used for machine pools without a platform configuration. |
+`apiVIP` | `api.<clusterdomain>` | The VIP to use for internal API communication. |
+`ingressVIP` | `test.apps.<clusterdomain>` | The VIP to use for ingress traffic. |
+`dnsVIP` | | The VIP to use for internal DNS communication. |
+
+##### VIP Settings
+
+The `apiVIP` and `ingressVIP` settings must either be provided or
+pre-configured in DNS so that the default names resolve correctly.
```
Comment (cybertron) [inline docs/user/metal/install_ipi.md]: Should we say what those names are?

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1518/

File: docs/user/metal/customization_ipi.md [hunk 5]
```diff
@@ -2,6 +2,23 @@
...
+`bootstrapProvisioningIP` | The second address on the provisioning network. `172.22.0.2` | The IP on the bootstrap VM where the provisioning services run while the control plane is being deployed. |
+`externalBridge` | `baremetal` | The name of the bridge of the hypervisor attached to the external network. |
+`provisioningBridge` | `provisioning` | The name of the bridge on the hypervisor attached to the provisioning network. |
+`provisioningNetworkCIDR` | `172.22.0.0/24` | The CIDR for the network to use for provisioning. |
+`provisioningDHCPExternal` | `false` | Flag indicating that DHCP for the provisioning network is managed outside of the cluster by existing infrastructure services. |
+`provisioningDHCPRange` | The tenth through the hundredth IP on the provisioning network. `172.22.0.10,172.22.0.100` | The IP range to use for hosts on the provisioning network. |
+`defaultMachinePlatform` | | The default configuration used for machine pools without a platform configuration. |
+`dnsVIP` | | The VIP to use for internal DNS communication. |
```
Comment (cybertron) [inline docs/user/metal/customization_ipi.md]: This is still duplicated.

File: docs/user/metal/install_ipi.md (hunk 4 above)
Comment (dhellmann) [reply docs/user/metal/install_ipi.md]: I added a reference to the table in the previous section that gives the names.

Comment (cybertron) [reply docs/user/metal/install_ipi.md]: Ah, I glossed over that. Thanks.

**Comment by metal3ci** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/1519/

**Pull request merged**